include/zephyr/devicetree.h

```diff
@@ -911,6 +913,95 @@
 	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
 		(DT_STRING_UPPER_TOKEN(node_id, prop)), (default_value))
 
+/**
+ * @brief Get a string-array item value as a token.
+ *
+ * This removes "the quotes" from string-valued item, and converts
+ * non-alphanumeric characters to underscores. That can be useful, for example,
+ * when programmatically using the value to form a C variable or code.
+ *
+ * DT_STRING_TOKEN_BY_IDX() can only be used for properties with
+ * string-array type.
+ *
+ * It is an error to use DT_STRING_TOKEN_BY_IDX() in other circumstances.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             prop = "f1", "F2";
+ *     };
+ *     n2: node-2 {
+ *             prop = "123 foo", "456 FOO";
+ *     };
+ *
+ * Example bindings fragment:
+ *
+ *     properties:
+ *       prop:
+ *         type: string-array
+ *
+ * Example usage:
+ *
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n1), prop, 0) // f1
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n1), prop, 1) // F2
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n2), prop, 0) // 123_foo
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n2), prop, 1) // 456_FOO
+ *
+ * For more notices see @ref DT_STRING_TOKEN.
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property string name
+ * @param idx the index to get
+ * @return the value of @p prop as a token, i.e. without any quotes
+ *         and with special characters converted to underscores
+ */
+#define DT_STRING_TOKEN_BY_IDX(node_id, prop, idx) \
+	DT_CAT4(node_id, _P_, prop##_IDX_##idx, _STRING_TOKEN)
```

Comment:
Use DT_CAT6 instead. See the comment above DT_CAT for an explanation of these macros.

Reply:
Done